### PR TITLE
ON-36370 # Updated TaskGroupInstance type

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: typescript/scheduledTasks.ts [hunk 1]
```diff
@@ -113,9 +113,16 @@ export type NewTaskGroup = {
...
   taskGroupId: number
-  taskGroupInstanceId: string
   label: string
 }
+
+export type EditedTaskGroupInstance = NewTaskGroupInstance & {
+  taskGroupInstanceId: string
+}
+
+export type TaskGroupInstance = EditedTaskGroupInstance & {
```
Comment (jdawg093) [inline typescript/scheduledTasks.ts:125]: Can you also add `TaskGroupInstance` to the auditing types please :)

File: typescript/scheduledTasks.ts [hunk 2]
```diff
@@ -69,17 +84,27 @@ export interface NewTask {
   swipeLeftActionId?: number
```
Comment (jdawg093) [inline typescript/scheduledTasks.ts:84]: actionIds needs to be a string array now, pleaaase and thank you

Comment (jdawg093) [reply typescript/scheduledTasks.ts:84]: `swipeLeftActionId` and `swipeRightActionId` also require this change to be a string

File: typescript/organisations.d.ts [hunk 3]
```diff
@@ -163,6 +163,7 @@ export type AuditRecordType =
   | 'CompletedTask'
   | 'ScheduledTaskAction'
   | 'ScheduledTaskGroup'
+  | 'ScheduledTaskGroupInstance'
```
Comment (divporter) [inline typescript/organisations.d.ts:166]: Do we still need this?

File: typescript/scheduledTasks.ts [hunk 4]
```diff
@@ -1,7 +1,12 @@
 import { MiscTypes } from '..'
 
+interface WithVersion {
+  versionId: number
+  createdAt: string
+}
 export type DayOfWeek = 'MON' | 'TUE' | 'WED' | 'THU' | 'FRI' | 'SAT' | 'SUN'
 
+// // Task Action // //
```
Comment (divporter) [inline typescript/scheduledTasks.ts:9]:
```suggestion
// Task Action //
```

File: typescript/scheduledTasks.ts [hunk 5]
```diff
@@ -13,24 +18,34 @@ type BaseTaskAction = {
...
+  taskActionId: string
+}
+export type ChangeStatusTaskAction = EditedChangeStatusAction & WithVersion
 
+// Final Actions //
 export type NewTaskAction = NewFormTaskAction | NewChangeStatusTaskAction
+export type EditedTaskAction = EditedFormTaskAction | EditedChangeStatusAction
 export type TaskAction = FormTaskAction | ChangeStatusTaskAction
 
+// // Task // //
```
Comment (divporter) [inline typescript/scheduledTasks.ts:48]:
```suggestion
// Task //
```

File: typescript/scheduledTasks.ts [hunk 6]
```diff
@@ -69,17 +84,27 @@ export interface NewTask {
...
 }
-export type Task = NewTask & MiscTypes.IdResource
 
+export type EditedTask = NewTask & {
+  taskId: string
+}
+
+export type Task = EditedTask & WithVersion
+
+// // Completed Task // //
```
Comment (divporter) [inline typescript/scheduledTasks.ts:94]:
```suggestion
// Completed Task //
```

File: typescript/scheduledTasks.ts [hunk 7]
```diff
@@ -97,6 +122,12 @@ export interface CompletedTask extends NewCompletedTask {
   id: string
 }
 
+// // Task Group // //
```
Comment (divporter) [inline typescript/scheduledTasks.ts:125]:
```suggestion
// Task Group //
```

**Pull request merged**